### ISSUE TITLE
Embedding annotations in the target document

### ISSUE THREAD

**Issue by azaroth42** [w3c/web-annotation#87]
Should have a recorded use case for this.
[rswick]

**Comment by tilgovi** [comment]
Can the dpub group help with this? Seems like footnotes and glosses in an EPUB book have this use case.

**Comment by csarven** [comment]
My use case is where 1) user marks some text in an HTML document - this becomes the target, 2) leaves an annotation - this becomes the body. I have something like the following:

```
# The target
<mark id="foo" about="[this:#foo]" typeof="dcmitype:Text">
  foo <em>bar</em> baz
</mark>

<a rel="oa:hasTarget" href="#foo">...</a>
<div rel="oa:hasBody">
  <div id="baz" about="[this:#baz]" typeof="oa:TextualBody" property="oa:text">
    <p>I agree that foo-bar-baz is as good as it gets.</p>
  </div>
</div>
```

I'm considering/looking for a property to label/describe the target resource. For example, with `oa:text` it'll be like:

```
<mark id="foo" about="[this:#foo]" typeof="dcmitype:Text" property="oa:text">
  foo <em>bar</em> baz
</mark>
```

but it could just as well be any other property, e.g., `schema:description`, which can actually provide the content of #foo.

**Comment by iherman** [comment]
> On 7 Nov 2015, at 01:06, Sarven Capadisli notifications@github.com wrote:
> 
> My use case is where 1) user marks some text in an HTML document - this becomes the target, 2) leaves an annotation - this becomes the body. I have something like the following:
> 
> # The target
> 
> <mark about="[this:#foo]" typeof="http://purl.org/dcmitype/Text">
>   foo <em>bar</em> baz
> </mark>
> 
> <a rel="oa:hasTarget" href="#foo">...</a>
> 
> <div rel="oa:hasBody">
>   <div about="[this:#baz]" typeof="oa:TextualBody" property="oa:text">
>     <p>I agree that foo-bar-baz is as good as it gets.</p>
>   </div>
> </div>
> 
> I'm considering/looking for a property to label/describe the target resource. For example, with oa:text it'll be like:
> 
> <mark about="[this:#foo]" typeof="http://purl.org/dcmitype/Text" property="oa:text">
>   foo <em>bar</em> baz
> </mark>
> but it could just as well be any other property, e.g., schema:description, which can actually provide the content of #foo.
> 
> I am sorry, @csarven, but I fail to understand what you want to achieve. In your example the last thing means that you have created an extra triple

```
<foo> oa:text "foo <em>bar</em> baz";
```

But the annotation will still have <foo> as its target. That extra triple is, in a sense, completely outside of the annotation structure. Can you explain what you want to achieve?

**Comment by azaroth42** [comment]
Can someone provide a proposal that would allow us to discuss and close the issue please?

Otherwise I propose "postpone".

**Comment by csarven** [comment]
1. The issue asks to document an embedded annotation in target. I have provided one. Have I misunderstood what the issue is asking for? What I was merely seeking was a way to "enrich" - if you want to call it that - the target by making the human-readable part as an RDF label. To simplify my example above, it covers cases where anyone, e.g., authors, editors, with the write access to say an HTML+RDFa document (e.g., the body and target are in the same document) can annotate.
2. @iherman If my example above is on track with this issue, then by my understanding, your statement "That extra triple is, in a sense, completely outside of the annotation structure" is in conflict with this issue :) Having said that, I did realize the importance of your point, and it made me re-question the whole thing on whether we want to have targets "annotated" at all. Or is it an exercise of just linking out to it i.e., hasTarget <IRI>.

**Comment by azaroth42** [comment]
@csarven - By proposal I mean: Is there something that needs to change in the model or protocol to allow the use case to happen, and if so, what?

**Comment by csarven** [comment]
@azaroth42 I would propose or at least entertain the idea of allowing the same features for http://www.w3.org/TR/annotation-model/#embedded-bodies for the target as we have for the body. In my specific example, I was sort of missing the oa:text (or any labeler), but I can see that the rest of what's available for embedded bodies is useful for the target as well.

**Comment by BigBlueHat** [comment]
So. I think things got tangled here. The issue title doesn't seem to state any need of putting the target _inside_ the annotation, but rather than reverse.

@csarven if you're wanting to include additional statements about the target in your RDFa, please do. :smile: I don't think anything in the model needs to change to support that. You're using RDF anyhow, so knock yourself out! :wink: 

However, I do think there's still a non-model topic here about how to embed annotations inside the target document...but that would depend on the format of the document (let's not assume it's HTML), etc.

Riffing off @tilgovi's DPUB / EPUB comment, we do have the use cases for this "embedded annotation" scenario and we _should_ for our own sanity limit the scope to a handful of formats and scenarios and map Web Annotation into the right location within those formats (EPUB, HTML, Link header references, MIME, package formats, etc)

Right now, that seems like good wiki exploration fodder with a possible NOTE as output and perhaps another serialization document but focused on RDFa (for HTML)--since HTML is likely the most frequently used format (for us) and @csarven's here to help use do that. :wink:

**Comment by azaroth42** [comment]
Given the amount of new work required, I propose `postpone`.  In particular, we don't have an html or other serialization document in which any of the above would be recorded.  I agree with @BigBlueHat that this is a good topic for strawpeople and use case descriptions in the wiki.

**Comment by iherman** [comment]
+1 to @azaroth42. I presumed postponed issues may be picked up if we work on a new charter, ie, we can come back to this later if and when we do that.

**Comment by shepazu** [comment]
Note that this is related to the notion of an HTML serialization of the annotation model, for use with footnotes and embedded annotations/comments. I have a rough prototype for something like this, but it needs work and a spec.

**Comment by iherman** [comment]
I have the impression that two different issues are mixed up here. 
1. What to do with embedding of annotations in the target document. That is the original issue, I believe, and I agree postponing it. It has not been seriously discussed in the group so far, it is a typical topic for the next possible incarnation of the group.
2. The issue raised by @csarven is, I believe, different. At present the model document allows for a more detailed characterization of the body. See many of the sections in [Section 3.2](https://w3c.github.io/web-annotation/model/wd/#bodies-and-targets) of the current latest draft. Eg, set the language, creator, media type, etc. I believe the issue/question is whether the same facilities should be provided to targets as well.

Considering @csarven: although I can see the point of doing this, eg, specifying that the target is a moving image, I am not sure those are annotation specific use cases. Using RDF (whether in JSON-LD or Turtle or RDFa) means that those extra properties can be added, of course, to any target, but I do not believe that it is the issue for the WA model to provide a fine characterization of the target; it is pretty reasonable to say that the focus is on the body which is, in some sense, the "active" part of an annotation. In other words, I do not see any modification on the model.

I would be o.k. if the document included a note somewhere in that section making it clear to the reader that application MAY use the properties like dctypes or dcterms to further characterize the target, too, but this document concentrates on the body only. Or something like that.

**Comment by csarven** [comment]
@ivan What I've raised touches both 1 and 2 as you've listed in https://github.com/w3c/web-annotation/issues/87#issuecomment-175630091 . The example I gave covers "footnotes and embedded annotations/comments" as @shepazu mentioned. In addition to that, I was hoping to have a clarification in the spec for oa:text so that it can be used by either body or target. This is not currently prevented, so that's all good.

Putting URLs with fragments aside: I think of the above as "internal" annotations, i.e., the body being in the same URL as target. An "external" annotation would be where body and target are different URLs.
